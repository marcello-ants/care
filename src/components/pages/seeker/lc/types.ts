import { GeneralProviderProfile } from '@/types/seeker';

export interface ProviderProfile extends GeneralProviderProfile {
  numberOfReviews: number;
  qualities?: ProviderQualities;
  services?: ProviderServices;
  firstName: string;
}

export interface ProviderQualities {
  certifiedNursingAssistant?: boolean;
  certifiedRegisterNurse?: boolean;
  certifiedTeacher?: boolean;
  childDevelopmentAssociate?: boolean;
  comfortableWithPets?: boolean;
  cprTrained?: boolean;
  crn?: boolean;
  doesNotSmoke?: boolean;
  doula?: boolean;
  earlyChildDevelopmentCoursework?: boolean;
  earlyChildhoodEducation?: boolean;
  firstAidTraining?: boolean;
  nafccCertified?: boolean;
  ownTransportation?: boolean;
  specialNeedsCare?: boolean;
  trustlineCertifiedCalifornia?: boolean;
}

export interface ProviderServices {
  carpooling?: boolean;
  craftAssistance?: boolean;
  errands?: boolean;
  groceryShopping?: boolean;
  laundryAssistance?: boolean;
  lightHousekeeping?: boolean;
  mealPreparation?: boolean;
  swimmingSupervision?: boolean;
  travel?: boolean;
}
