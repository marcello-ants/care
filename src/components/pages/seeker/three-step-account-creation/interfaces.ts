import { VerticalsAbbreviation } from '@/constants';
import { SeekerHKState } from '@/types/seekerHK';
import { SeekerPCState } from '@/types/seekerPC';
import { SeekerTUState } from '@/types/seekerTU';
import { SeekerCCState } from '@/types/seekerCC';
import { Dispatch, SetStateAction } from 'react';

export type SetSeekerNameAction =
  | 'cc_setSeekerName'
  | 'hk_setSeekerName'
  | 'pc_setSeekerName'
  | 'tu_setSeekerName';

export type SeekerVerticalState = SeekerCCState | SeekerHKState | SeekerPCState | SeekerTUState;

export interface AccountCreationFormPropsBase {
  vertical: VerticalsAbbreviation;
  nextPageURL: string;
}

export interface AccountCreationNameFormProps extends AccountCreationFormPropsBase {
  verticalState: SeekerVerticalState;
}

export interface AccountCreationGeneralFormProps extends AccountCreationFormPropsBase {
  setFormikSubmittingFalse: () => void;
  setErrorMessage: Dispatch<SetStateAction<string | null>>;
}

export interface FormNameValues {
  firstName: string;
  lastName: string;
  email: string;
  howDidYouHearAboutUs?: string;
}
