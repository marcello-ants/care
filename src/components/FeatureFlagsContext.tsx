import { createContext, useContext, ReactNode, useState } from 'react';
import { LDEvaluationDetail } from 'launchdarkly-node-server-sdk';

interface FeatureFlagsProviderProps {
  children: ReactNode;
  flags: FeatureFlags;
}

export type featureFlagsState = {
  flags: FeatureFlags;
};

export type FeatureFlags = Record<string, LDEvaluationDetail | undefined>;

const initialState: featureFlagsState = {
  flags: {},
};

const FeatureFlagsContext = createContext<featureFlagsState>(initialState);

export const FeatureFlagsProvider = ({ children, flags }: FeatureFlagsProviderProps) => {
  const [state] = useState({ flags });
  return <FeatureFlagsContext.Provider value={state}>{children}</FeatureFlagsContext.Provider>;
};

export const useFeatureFlags = () => useContext(FeatureFlagsContext);
