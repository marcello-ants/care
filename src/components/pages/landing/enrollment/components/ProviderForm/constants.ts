export const PROVIDER_SERVICES_DESKTOP = [
  {
    description: 'Child care',
    label: 'Child care',
    value: 'CHILD_CARE',
  },
  {
    description: 'Senior care',
    label: 'Senior care',
    value: 'SENIOR_CARE',
  },
  {
    description: 'Housekeeping',
    label: 'Housekeeping',
    value: 'HOUSEKEEPING',
  },
  {
    description: 'Pet care',
    label: 'Pet care',
    value: 'PET_CARE',
  },
  {
    description: 'Tutoring',
    label: 'Tutoring',
    value: 'TUTORING',
  },
  {
    description: 'Special needs care',
    label: 'Special needs care',
    value: 'SPECIAL_NEEDS',
  },
  {
    description: 'Errands & odd jobs',
    label: 'Errands & odd jobs',
    value: 'GENERIC',
  },
];

export const PROVIDER_SERVICES_MOBILE = [
  {
    description: 'Child care',
    label: 'Child care',
    value: 'CHILD_CARE',
  },
  {
    description: 'Senior care',
    label: 'Senior care',
    value: 'SENIOR_CARE',
  },
  {
    description: 'Pet care',
    label: 'Pet care',
    value: 'PET_CARE',
  },
  {
    description: 'Tutoring',
    label: 'Tutoring',
    value: 'TUTORING',
  },
  {
    description: 'Special needs care',
    label: 'Special needs care',
    value: 'SPECIAL_NEEDS',
  },
  {
    description: 'Housekeeping',
    label: 'Housekeeping',
    value: 'HOUSEKEEPING',
  },
  {
    description: 'Errands & odd jobs',
    label: 'Errands & odd jobs',
    value: 'GENERIC',
  },
];

export const DROPOFF_URLS = {
  CHILD_CARE: '/swf/ccOptimizedFlow',
  SENIOR_CARE: '/app/enrollment/provider/sc ',
  PET_CARE: '/swf/pcOptimizedFlow',
  HOUSEKEEPING: '/swf/hkOptimizedFlow',
  TUTORING: '/swf/tuOptimizedFlow',
  SPECIAL_NEEDS: '/swf/snOptimizedFlow',
  GENERIC: '/swf/cgOptimizedFlow',
  AU_PAIR: '',
  CARE_GIGS: '',
};
