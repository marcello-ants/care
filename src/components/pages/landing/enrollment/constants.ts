export const SITTER_FORM = {
  providerServices: {
    title: 'What Services do you offer?',
    id: 'providerService',
    options: [
      {
        value: 'CHILD_CARE',
        text: 'Babysitter',
      },
      {
        value: 'CHILD_CARE',
        text: 'Nanny',
      },
      {
        value: 'SPECIAL_NEEDS',
        text: 'Special Needs Provider',
      },
      {
        value: 'TUTORING',
        text: 'Tutor',
      },
      {
        value: 'TUTORING',
        text: 'Private Lesson Instructor',
      },
      {
        value: 'SENIOR_CARE',
        text: 'Senior Care Provider',
      },
      {
        value: 'PET_CARE',
        text: 'Pet Care Provider',
      },
      {
        value: 'HOUSEKEEPING',
        text: 'Housekeeper',
      },
      {
        value: 'GENERIC',
        text: 'Errands &amp; Odd Jobs Provider',
      },
    ],
  },
  providerEmail: {
    id: 'email',
    placeholder: 'Email',
  },
  providerPassword: {
    id: 'password',
    placeholder: 'Password',
  },
  providerZip: {
    id: 'providerZip',
    placeholder: 'ZIP',
  },
};

export const GENERIC_FORM = {
  howDidYouHearAboutUs: {
    title: 'How did you hear about us?',
    id: 'howDidYouHearAboutUs',
    options: [
      {
        value: 'FACEBOOK',
        text: 'Facebook',
      },
      {
        value: 'BANNER_AD',
        text: 'Banner Ad',
      },
      {
        value: 'TV_AD',
        text: 'TV Ad',
      },
      {
        value: 'OTHER_SOCIAL_MEDIA',
        text: 'Other Social Media (Twitter, Instagram, Pinterest, LinkedIn)',
      },
      {
        value: 'PRESS_COVERAGE',
        text: 'Press Coverage (News, Magazine, Blog)',
      },
      {
        value: 'FRIENDS_OR_FAMILY',
        text: 'Friends or Family',
      },
      {
        value: 'ONLINE_VIDEO',
        text: 'Online Video (Hulu, Youtube)',
      },
      {
        value: 'SEARCH_ENGINE',
        text: 'Search Engine (Google, Bing)',
      },
      {
        value: 'PARENTING_GROUP',
        text: 'Parenting Group or Forum',
      },
      {
        value: 'OTHER',
        text: 'Other',
      },
    ],
  },
};
