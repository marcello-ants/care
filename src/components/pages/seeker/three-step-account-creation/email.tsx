import getConfig from 'next/config';
import * as yup from 'yup';
import { FormikProvider, useFormik } from 'formik';
import { Box, Button, Grid, makeStyles } from '@material-ui/core';
import { Banner, Typography } from '@care/react-component-lib';
import userEmailStore from '@/lib/UserEmailStore';
import useProviderCount from '@/components/hooks/useProviderCount';
import useFlowNavigation from '@/components/features/flowNavigation/useFlowNavigation';
import OverlaySpinner from '@/components/OverlaySpinner';
import EmailField from '@/components/features/accountCreation/EmailField';
import { ServiceType } from '@/__generated__/globalTypes';
import { logCareEvent } from '@/utilities/amplitudeAnalyticsHelper';

const {
  publicRuntimeConfig: { CZEN_GENERAL_LOGIN },
} = getConfig();

interface FormValues {
  email: string;
}

const useStyles = makeStyles(() => ({
  inputContainer: {
    '& > div': {
      padding: 0,
    },
  },
}));

const VERTICAL_CAREGIVERS: { [key: string]: string } = {
  [ServiceType.HOUSEKEEPING]: 'housekeepers',
  [ServiceType.PET_CARE]: 'pet caregivers',
  [ServiceType.TUTORING]: 'tutors',
};

interface AccountCreationEmailFormProps {
  serviceType: ServiceType;
}

function AccountCreationEmailForm({ serviceType }: AccountCreationEmailFormProps) {
  const classes = useStyles();
  const { numOfProviders, displayProviderMessage } = useProviderCount(serviceType);
  const { goNext } = useFlowNavigation();

  const validationSchema = yup.object({
    email: yup.string().required('Email is required'),
  });

  const handleSubmission = (values: FormValues) => {
    userEmailStore.setEmail(values.email);
    logCareEvent('Member Enrolled', 'Email', {
      caregiverCount: numOfProviders,
      member_type: 'Seeker',
    });
    goNext();
  };

  const formik = useFormik<FormValues>({
    initialValues: {
      email: userEmailStore.getEmail(),
    },
    validationSchema,
    onSubmit: handleSubmission,
  });

  if (formik.isSubmitting) {
    return <OverlaySpinner isOpen wrapped />;
  }

  return (
    <FormikProvider value={formik}>
      {displayProviderMessage && (
        <Box mb={3} ml={1} mr={1}>
          <Banner type="information" width="100%" roundCorners>
            <Typography variant="body2" data-testid="housekeepers-nearby">
              Nice! {numOfProviders} {VERTICAL_CAREGIVERS[serviceType]} near you!
            </Typography>
          </Banner>
        </Box>
      )}
      <Grid container>
        <Box pl={1} pr={1}>
          <Typography variant="h2">Create a free account</Typography>
          <Box mt={1}>
            <Typography variant="body2">
              See {VERTICAL_CAREGIVERS[serviceType]} who match your needs. It only takes a few
              seconds.
            </Typography>
          </Box>
          <Grid item xs={12} className={classes.inputContainer}>
            <EmailField
              id="email"
              name="email"
              label="Email address"
              czenGeneralLoginUrl={CZEN_GENERAL_LOGIN}
              showErrorWhenTouched
            />
          </Grid>
          <Grid item xs={12}>
            <Box mt={5}>
              <Button
                color="primary"
                variant="contained"
                size="large"
                fullWidth
                onClick={formik.submitForm}
                tabIndex={0}>
                Next
              </Button>
            </Box>
          </Grid>
        </Box>
      </Grid>
    </FormikProvider>
  );
}

export default AccountCreationEmailForm;
