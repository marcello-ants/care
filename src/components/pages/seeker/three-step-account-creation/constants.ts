import * as yup from 'yup';
import { ServiceType } from '@/__generated__/globalTypes';
import { VerticalsAbbreviation } from '@/constants';

export const validationSchema = yup.object({
  firstName: yup
    .string()
    .min(2, 'First name needs to have at least two letters.')
    .required('First name is required'),
  lastName: yup
    .string()
    .min(2, 'Last name needs to have at least two letters.')
    .required('Last name is required'),
});

export const SERVICE_TYPE_BY_VERTICAL: { [key: string]: ServiceType } = {
  HK: ServiceType.HOUSEKEEPING,
  PC: ServiceType.PET_CARE,
  TU: ServiceType.TUTORING,
  CC: ServiceType.CHILD_CARE,
  DC: ServiceType.CHILD_CARE,
  IB: ServiceType.CHILD_CARE,
  SC: ServiceType.SENIOR_CARE,
};

export const ERROR_CREATING_ACCOUNT = 'An error occurred creating your account, please try again.';

export const FACEBOOK_VERTICALS_ALLOWANCE = ['CC'];
export const HDYHAU_VERTICALS_ALLOWANCE: VerticalsAbbreviation[] = ['HK', 'TU', 'PC', 'DC'];
