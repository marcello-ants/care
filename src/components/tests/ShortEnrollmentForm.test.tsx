import { AppState } from '@/types/app';
import { MockedResponse } from '@apollo/client/testing';
import { fireEvent, screen, waitFor, within } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { initialAppState } from '@/state';
import { ServiceType } from '@/__generated__/globalTypes';
import AuthService from '@/lib/AuthService';
import { AnalyticsHelper } from '@/utilities/analyticsHelper';
import {
  GET_ZIP_CODE_SUMMARY,
  SEEKER_CREATE,
  VALIDATE_MEMBER_EMAIL,
} from '@/components/request/GQL';
import { preRenderPage } from '@/__setup__/testUtil';
import { ERROR_CREATING_ACCOUNT } from '@/components/pages/seeker/three-step-account-creation/constants';
import { SHORT_ENROLLMENT_ROUTES } from '@/constants';

import { FeatureFlags } from '../FeatureFlagsContext';
import ShortEnrollmentForm from '../ShortEnrollmentForm';

let buttonJoinNow: HTMLElement;
let textboxEmail: HTMLElement;
let textboxFirstName: HTMLElement;
let textboxLastName: HTMLElement;
let zipcodeInput: HTMLElement;
let buttonCareKind: HTMLElement;

jest.mock('@/lib/AuthService');
const AuthServiceMock = AuthService as jest.Mock;

const analyticsMock = jest
  .spyOn(AnalyticsHelper, 'logEvent')
  .mockImplementation(() => Promise.resolve());

const redirectLoginSpy = jest.fn();
AuthServiceMock.mockImplementation(() => {
  return {
    redirectLogin: redirectLoginSpy,
  };
});

const initialState: AppState = {
  ...initialAppState,
  seeker: {
    ...initialAppState.seeker,
    zipcode: '',
  },
};

const TEST_DATA_SUCCESS_FIRST_NAME = 'John';
const TEST_DATA_SUCCESS_LAST_NAME = 'Doe';
const TEST_DATA_SUCCESS_EMAIL = 'email@success.com';
const TEST_DATA_SUCCESS_ZIP_CODE = '02453';

const getZipcodeSummarySuccessMock = {
  request: {
    query: GET_ZIP_CODE_SUMMARY,
    variables: {
      zipcode: TEST_DATA_SUCCESS_ZIP_CODE,
    },
  },
  result: {
    data: {
      getZipcodeSummary: {
        __typename: 'ZipcodeSummary',
        city: 'Waltham',
        state: 'MA',
        zipcode: TEST_DATA_SUCCESS_ZIP_CODE,
        latitude: 1,
        longitude: 1,
      },
    },
  },
};

const seekerCreateWithoutReferrerCookieSuccessMock = {
  request: {
    query: SEEKER_CREATE,
    variables: {
      input: {
        email: TEST_DATA_SUCCESS_EMAIL,
        firstName: TEST_DATA_SUCCESS_FIRST_NAME,
        lastName: TEST_DATA_SUCCESS_LAST_NAME,
        serviceType: ServiceType.CHILD_CARE,
        zipcode: TEST_DATA_SUCCESS_ZIP_CODE,
        howDidYouHearAboutUs: undefined,
        careDate: null,
        referrerCookie: undefined,
      },
    },
  },
  result: {
    data: {
      seekerCreate: {
        __typename: 'SeekerCreateSuccess',
        authToken: 'pPMAzhsSOJx5ZxZAy_e27*n28NNwBug4D0III1hM1Dw.',
        memberId: '4747',
      },
    },
  },
};

const seekerCreateWithoutReferrerCookieFailureMock = {
  request: {
    query: SEEKER_CREATE,
    variables: {
      input: {
        email: TEST_DATA_SUCCESS_EMAIL,
        firstName: TEST_DATA_SUCCESS_FIRST_NAME,
        lastName: TEST_DATA_SUCCESS_LAST_NAME,
        serviceType: ServiceType.CHILD_CARE,
        zipcode: TEST_DATA_SUCCESS_ZIP_CODE,
        howDidYouHearAboutUs: undefined,
        careDate: null,
        referrerCookie: undefined,
      },
    },
  },
  result: {
    data: {
      seekerCreate: {
        errors: [
          {
            __typename: 'MemberCreateError',
            message: ERROR_CREATING_ACCOUNT,
          },
        ],
        __typename: 'SeekerCreateError',
      },
    },
  },
};

const emailValidationFailureMock = {
  request: {
    query: VALIDATE_MEMBER_EMAIL,
    variables: {
      email: '',
    },
  },
  result: {
    data: {
      validateMemberEmail: { errors: [{ message: 'Please enter a valid email.' }] },
    },
  },
};

const emailValidationSuccessMock = {
  request: {
    query: VALIDATE_MEMBER_EMAIL,
    variables: {
      email: TEST_DATA_SUCCESS_EMAIL,
    },
  },
  result: {
    data: {
      validateMemberEmail: { errors: [] },
    },
  },
};

async function renderPage(
  mocks: MockedResponse[] = [],
  ldFlags: FeatureFlags = {},
  state: AppState = initialState,
  pathname: string = SHORT_ENROLLMENT_ROUTES.INDEX,
  nextPageURL: string = 'password'
) {
  const { renderFn } = preRenderPage({ mocks, flags: ldFlags, appState: state, pathname });

  return renderFn(<ShortEnrollmentForm nextPageURL={nextPageURL} />);
}

const getElements = () => {
  textboxFirstName = screen.getByRole('textbox', { name: 'First name' });
  textboxLastName = screen.getByRole('textbox', { name: 'Last name' });
  textboxEmail = screen.getByRole('textbox', { name: 'Email address' });
  zipcodeInput = screen.getByLabelText('ZIP code');
  buttonJoinNow = screen.getByRole('button', { name: 'Join free now' });
  buttonCareKind = screen.getByRole('button', { name: 'What type of care do you need?' });
};

const enterFormDataAndTriggerSubmit = async () => {
  fireEvent.mouseDown(buttonCareKind);
  const listbox = within(screen.getByRole('listbox'));
  fireEvent.click(listbox.getByText('Child care'));
  await waitFor(() => {
    expect(screen.getByText('Child care')).toBeInTheDocument();
  });

  userEvent.type(textboxEmail, TEST_DATA_SUCCESS_EMAIL);
  fireEvent.blur(textboxEmail);
  await waitFor(() => {
    expect(textboxEmail).toBeValid();
  });

  userEvent.type(textboxFirstName, TEST_DATA_SUCCESS_FIRST_NAME);
  fireEvent.blur(textboxFirstName);
  await waitFor(() => {
    expect(textboxFirstName).toBeValid();
  });

  userEvent.type(textboxLastName, TEST_DATA_SUCCESS_LAST_NAME);
  fireEvent.blur(textboxLastName);
  await waitFor(() => {
    expect(textboxLastName).toBeValid();
  });

  fireEvent.change(zipcodeInput, { target: { value: TEST_DATA_SUCCESS_ZIP_CODE } });
  fireEvent.blur(zipcodeInput);

  await waitFor(() => {
    expect(screen.getByText('Waltham, MA')).toBeInTheDocument();
  });

  userEvent.click(buttonJoinNow);
};

describe('ShortEnrollmentForm', () => {
  it('matches snapshot', async () => {
    const { asFragment } = await renderPage();
    expect(asFragment()).toMatchSnapshot();
  });

  it('shows validation errors', async () => {
    await renderPage([emailValidationFailureMock]);
    getElements();
    userEvent.click(buttonJoinNow);
    expect(await screen.findByText(/Please select an option/)).toBeInTheDocument();
    expect(await screen.findByText(/First name is required/)).toBeInTheDocument();
    expect(await screen.findByText(/Last name is required/)).toBeInTheDocument();
    expect(await screen.findByText(/Please enter a valid email./)).toBeInTheDocument();
    expect(await screen.findByText(/Enter a valid ZIP code/)).toBeInTheDocument();
  });

  it('submits form successfully, logs events and gets redirected', async () => {
    renderPage([
      emailValidationSuccessMock,
      getZipcodeSummarySuccessMock,
      seekerCreateWithoutReferrerCookieSuccessMock,
    ]);
    getElements();

    await enterFormDataAndTriggerSubmit();

    await waitFor(() => {
      expect(analyticsMock).toHaveBeenCalledWith({
        name: 'Member Enrolled',
        data: {
          vertical: 'Childcare',
          final_step: true,
          enrollment_step: 'SingleEnrollment',
          cta_clicked: 'Join free now',
          enrollment_flow: 'SingleEnrollment',
          memberType: 'seeker',
        },
      });
    });

    await await waitFor(
      () =>
        expect(redirectLoginSpy).toHaveBeenCalledWith(
          '/password',
          'pPMAzhsSOJx5ZxZAy_e27*n28NNwBug4D0III1hM1Dw.'
        ),
      {
        timeout: 2000,
      }
    );
  });

  it('shows error message if the form submission is unsuccessful', async () => {
    renderPage([
      emailValidationSuccessMock,
      getZipcodeSummarySuccessMock,
      seekerCreateWithoutReferrerCookieFailureMock,
    ]);
    getElements();

    await enterFormDataAndTriggerSubmit();
    await waitFor(() => {
      expect(
        screen.getByText('An error occurred creating your account, please try again.')
      ).toBeInTheDocument();
    });
  });
});
