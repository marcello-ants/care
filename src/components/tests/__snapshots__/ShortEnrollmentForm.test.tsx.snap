// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShortEnrollmentForm matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiFormControl-root makeStyles-selectContainer-3"
      >
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline no-value MuiInputBase-formControl MuiInput-formControl"
        >
          <div
            aria-haspopup="listbox"
            aria-labelledby="mui-component-select-vertical"
            class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
            id="mui-component-select-vertical"
            role="button"
            tabindex="-1"
          >
            What type of care do you need?
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput"
            name="vertical"
            tabindex="-1"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-root-19"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18.2 8a.91.91 0 011.2 0 .91.91 0 010 1.2L12.6 16a.91.91 0 01-1.2 0L4.6 9.2a.91.91 0 010-1.2.91.91 0 011.2 0l6.2 6.2z"
              fill="#223f5f"
            />
            <path
              d="M12 16.37a1 1 0 01-.71-.26l-6.8-6.8a1.09 1.09 0 010-1.42 1.09 1.09 0 011.42 0L12 14l6.09-6.1a1.09 1.09 0 011.42 0 1.09 1.09 0 010 1.42l-6.8 6.8a1 1 0 01-.71.25zM5.2 7.93a.69.69 0 00-.49.18.76.76 0 000 1l6.8 6.8a.78.78 0 001 0l6.8-6.8a.76.76 0 000-1 .76.76 0 00-1 0L12 14.41l-.11-.1-6.2-6.2a.69.69 0 00-.49-.18z"
              fill="#223f5f"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
      >
        <div
          class="MuiFormControl-root MuiTextField-root makeStyles-inlineTextFieldOverride-1 makeStyles-inline-6"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink="false"
            for="firstName"
            id="firstName-label"
          >
            First name
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputHiddenLabel"
              id="firstName"
              name="firstName"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
      >
        <div
          class="MuiFormControl-root MuiTextField-root makeStyles-inlineTextFieldOverride-1 makeStyles-inline-6"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink="false"
            for="lastName"
            id="lastName-label"
          >
            Last name
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputHiddenLabel"
              id="lastName"
              name="lastName"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiFormControl-root MuiTextField-root makeStyles-inlineTextFieldOverride-1 makeStyles-inline-6"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink="false"
          for="email"
          id="email-label"
        >
          Email address
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            id="email"
            name="email"
            type="email"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiGrid-root makeStyles-zipInput-2 MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          class="MuiFormControl-root MuiTextField-root makeStyles-inlineTextFieldOverride-10 makeStyles-inlineTextFieldOverride-12 makeStyles-inline-6"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink="false"
            for="zipCode"
            id="zipCode-label"
          >
            ZIP code
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
              id="zipCode"
              name="zipCode"
              type="number"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiBox-root MuiBox-root-13"
      >
        <div
          class="MuiFormControl-root makeStyles-selectContainer-14"
        >
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline no-value MuiInputBase-formControl MuiInput-formControl"
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="source"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
              id="source"
              role="button"
              tabindex="-1"
            >
              How did you hear about us?
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              name="howDidYouHearAboutUs"
              tabindex="-1"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-root-20"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M18.2 8a.91.91 0 011.2 0 .91.91 0 010 1.2L12.6 16a.91.91 0 01-1.2 0L4.6 9.2a.91.91 0 010-1.2.91.91 0 011.2 0l6.2 6.2z"
                fill="#223f5f"
              />
              <path
                d="M12 16.37a1 1 0 01-.71-.26l-6.8-6.8a1.09 1.09 0 010-1.42 1.09 1.09 0 011.42 0L12 14l6.09-6.1a1.09 1.09 0 011.42 0 1.09 1.09 0 010 1.42l-6.8 6.8a1 1 0 01-.71.25zM5.2 7.93a.69.69 0 00-.49.18.76.76 0 000 1l6.8 6.8a.78.78 0 001 0l6.8-6.8a.76.76 0 000-1 .76.76 0 00-1 0L12 14.41l-.11-.1-6.2-6.2a.69.69 0 00-.49-.18z"
                fill="#223f5f"
              />
            </svg>
          </div>
          <p
            class="MuiFormHelperText-root"
          >
            <span>
              â€‹
            </span>
          </p>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-justify-content-xs-center"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
      >
        <div
          class="MuiBox-root MuiBox-root-16"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-fullWidth"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Join free now
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
